sr_data_visualization/src/sr_data_visualization/data_plot.py

```diff
@@ -91,17 +91,22 @@ def timerEvent(self):
 
         self.replot()
 
-    def plot_data(self, plot):
+    def plot_data(self, plot, side, new_sub=True):
+        self._topic_name = self._topic_name.replace("lh", side).replace("rh", side)
+
+        if self._subscriber:
+            self._subscriber.unregister()
+            self.timer.stop()
```

Comment:
should the timer stop when we get out of plot condition? i think it was like that before?

Reply:
Yes, before the stop was when plot was false and the subscriber existed. I added the stop at first because I unregister the subsription so I thought there was not any point in continuing updating the timer. I also unregister the subscription there to create a new one for a different side.